Shouldly for net451

Please add dedicated net451 target without CallerMemberNamePolyfill because when I'm using net40 Shouldly with net451 project I'm getting warnings like

CSC : warning CS1685: The predefined type 'System.Runtime.CompilerServices.CallerMemberNameAttribute' is defined in multiple assemblies in the global alias; using definition from 'c:\Program Files (x86)\Reference Assemblies\Microsoft\Framework\.NETFramework\v4.5.1\mscorlib.dll'

--- a/build.cake
+++ b/build.cake
@@ -1,4 +1,5 @@
 #tool "nuget:?package=GitReleaseNotes"
+#tool nuget:?package=GitVersion.CommandLine
 
 var target = Argument("target", "Default");
 var shouldlyProj = "./src/Shouldly/project.json";
@@ -14,7 +15,9 @@ Task("Clean")
 
 Task("Restore")
     .Does(() => {
-        NuGetRestore("./src/Shouldly.sln");
+        NuGetRestore("./src/Shouldly.sln", new NuGetRestoreSettings{
+            MSBuildVersion = NuGetMSBuildVersion.MSBuild14
+        });
     });
 
 GitVersion versionInfo = null;

--- a/src/Shouldly.Tests/ConventionTests/ApprovePublicAPI.ShouldlyApi.approved.cs
+++ b/src/Shouldly.Tests/ConventionTests/ApprovePublicAPI.ShouldlyApi.approved.cs
@@ -1,5 +1,5 @@
 [assembly: System.Runtime.InteropServices.ComVisibleAttribute(false)]
-[assembly: System.Runtime.Versioning.TargetFrameworkAttribute(".NETFramework,Version=v4.0")]
+[assembly: System.Runtime.Versioning.TargetFrameworkAttribute(".NETFramework,Version=v4.5.1")]
 
 namespace Shouldly
 {

--- a/src/Shouldly/project.json
+++ b/src/Shouldly/project.json
@@ -1,4 +1,4 @@
-﻿{
+{
     "description": "Shouldly - Assertion framework for .NET. The way asserting *Should* be",
     "frameworks": {
         "net35": {
@@ -24,6 +24,19 @@
                     "Async",
                     "ExpressionTrees",
                     "CallerMemberNamePolyfill"
+                ]
+            }
+        },
+        "net451": {
+            "buildOptions": {
+                "keyFile": "sn.snk",
+                "define": [
+                    "ShouldMatchApproved",
+                    "StackTrace",
+                    "Dynamic",
+                    "Serializable",
+                    "Async",
+                    "ExpressionTrees"
                 ]
             }
         },

